Syntect fails to build when combined with serde

I ran into this problem with cobalt.  I hacked around it by [disabling the default features I didn't need](https://github.com/cobalt-org/cobalt.rs/commit/3825a7e319f90a033eb20a3c967bb4d07cbdb355#diff-80398c5faae3c069e4e6aa2ed11b28c0).

Example error
```
error[E0308]: mismatched types
  --> /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/syntect-1.7.0/src/parsing/yaml_load.rs:90:34
   |
90 |         if let Ok(map) = get_key(h, "variables", |x| x.as_hash()) {
   |                                  ^ expected struct `std::collections::BTreeMap`, found struct `linked_hash_map::LinkedHashMap`
   |
   = note: expected type `&std::collections::BTreeMap<yaml_rust::Yaml, yaml_rust::Yaml>`
              found type `&linked_hash_map::LinkedHashMap<yaml_rust::Yaml, yaml_rust::Yaml>`
   = help: here are some functions which might fulfill your needs:
           - .hasher()
```

In case there is something weird we are doing with syntect
- [where its used](https://github.com/cobalt-org/cobalt.rs/blob/master/src/syntax_highlight.rs)
- [Cargo.toml](https://github.com/cobalt-org/cobalt.rs/blob/master/Cargo.toml)
- [lib.rs](https://github.com/cobalt-org/cobalt.rs/blob/master/src/lib.rs)

--- a/src/parsing/yaml_load.rs
+++ b/src/parsing/yaml_load.rs
@@ -1,7 +1,8 @@
 use super::scope::*;
 use super::syntax_definition::*;
 use yaml_rust::{YamlLoader, Yaml, ScanError};
-use std::collections::{HashMap, BTreeMap};
+use yaml_rust::yaml::Hash;
+use std::collections::HashMap;
 use onig::{self, Regex, Captures};
 use std::rc::Rc;
 use std::cell::RefCell;
@@ -30,7 +31,7 @@ pub enum ParseSyntaxError {
     TypeMismatch,
 }
 
-fn get_key<'a, R, F: FnOnce(&'a Yaml) -> Option<R>>(map: &'a BTreeMap<Yaml, Yaml>,
+fn get_key<'a, R, F: FnOnce(&'a Yaml) -> Option<R>>(map: &'a Hash,
                                                     key: &'static str,
                                                     f: F)
                                                     -> Result<R, ParseSyntaxError> {
@@ -132,7 +133,7 @@ impl SyntaxDefinition {
         Ok(defn)
     }
 
-    fn parse_contexts(map: &BTreeMap<Yaml, Yaml>,
+    fn parse_contexts(map: &Hash,
                       state: &mut ParserState)
                       -> Result<HashMap<String, ContextPtr>, ParseSyntaxError> {
         let mut contexts = HashMap::new();
@@ -246,7 +247,7 @@ impl SyntaxDefinition {
         })
     }
 
-    fn parse_match_pattern(map: &BTreeMap<Yaml, Yaml>,
+    fn parse_match_pattern(map: &Hash,
                            state: &mut ParserState)
                            -> Result<MatchPattern, ParseSyntaxError> {
         let raw_regex = try!(get_key(map, "match", |x| x.as_str()));

